README.md

# A3_vemf_re_reloaded

Vemf_reloaded was originally made by IT07 and last commit on vemf_reloaded was in 2017.
I really like this mission script, but in my opinion it was missing some essential features compared to other mission systems.

So I decided to add and fix some of the stuff for vemf.

Check the Changelog for fixes and new stuff.

**IMPORTANT !!!  If you alreaduy have vemf installed it´s important to overwrite your existing files with the vemf_re_reloaded.
			   Then reconfigure the config.cpp to your need.**


If you have any questions/bug reports or ideas for vemf_re_reloaded then find me on discord Porkeld#1506 


All credit goes to **IT07** for coding this and also a big **THANK YOU** goes to **Ghostrider [GRG]** for helping and always being helpfull when needed :-)







a3_vemf_reloaded a.k.a. VEMFr
a complex mission system for ArmA3 Exile/Epoch Mod
ABOUT
VEMFr stands for "VEMF Reloaded". VEMF stands for "Vampire's Epoch Mission Framework".
VEMFr started as VEMF, under my ( IT07 ) continued development.
Initially, VEMF was obviously developed by TheVampire until he discontinued the project.
That is when I took over the project until TheVampire decided to continue VEMF himself.
Then I decided to also continue my work on VEMF but under a new name: VEMFr. And that is how VEMFr was born.


**[ CREDITS ]**
*TheVampire* ( the creator of VEMF )


Changelog.txt

v0.667

Fixed :   	Error in rpt.  removeAllEventHandlers 'MPKilled' } >

Fixed : 	bugs regarding tws blacklist

Fixed : 	Baseattack ai can also spawn mortars.

Fixed :		Typo in config.

Fixed : 	Issue with static ca50 when running epoch.


version 0.666


Fixed :    	Apex classes was not working. Added new check for apex and added "ApexEnabled =" in settings.
		Now all 4 factions will spawn.
			
Fixed : 	AI spawned without scopes on rifles. code for attacthment has been reworked.

Fixed : 	Weapons in lootcrates now spawns with correct ammo.

Fixed : 	Not all lightsources in cities was turned off when "disable street lights = yes"   Now invaded cities are completely 			black.

Fixed : 	Special forces had a grey color marker wich could be difficult to see if spawned on a spawn point.
		Made it pink! ;-)

Fixed : 	Cleaned up config and sorted some settings.

Added :		Possibility of spawning a vehicle patrol.

Added :		Vehicle patrol in invaded areas and spawn chance for both heli and vehicle in percentage.

Added :		Lights and new lightsource to heli patrol during night. Heli will now light up the ground below. If using a hellccat Ai 		will now use searchlight to track player.

added : 	Added some locations to blacklist that did not make sense.

Added :     	Mortars and chance in percentage for AI to be equipped with mortars - THIS is exciting.. AI will setup a mortar and 			bombard the player..... 

Added :		Class for AI loadout percentage in cfg :
			
			class aiLoadoutChance  // AI loadout chance
						
				ai_nighttime_accessory_chance =  // AI will (if equipped with a flashlight) light you up during 					nighttime.
				ai_scope_chance = 100;
				ai_muzzle_chance = 100;  //Select random muzzle
				ai_bipod_chance = 100;	 // Select random pointer
				ai_pointer_chance = 100; // Select random bipod
				ai_nighttime_NVGoggles_chance = 0;	// Percentage chance that AI will have a 							 NVGoggles spawned during nighttime


Added :		inventory loadout array for AI. Vemf AI´s had no loot in their inventory.

				Items[] = { "Exile_Item_DuctTape", "Exile_Item_CanOpener" };

				explosives[] = { "MiniGrenade", "MiniGrenade", "HandGrenade", "HandGrenade"  };
				
				medkits[] = { "Exile_item_Bandage", "Exile_Item_InstaDoc", "Exile_Item_Vishpirin" };

				food[] = { "Exile_Item_Moobar", "Exile_Item_BeefParts", "Exile_Item_EMRE" };

				binoc[] = { "Rangefinder", "Binocular" };

Added : 	added option to configure scope and attachments (muzzle + bipods) slots in config.    

				scopSlotsMax = 20; // Maximum number of scope slots in each loot crate
				scopSlotsMin = 10; // Minimum number of scope slots in each loot crate
			
added :		seperated array for muzzles and bipods in crateloot. 			


///   Not looked at yet.

BAseattack
Simplepatrol





VEMFr_client/gui/RscDisplayBaseAttack.hpp
class RscDisplayBaseAttack
{
   duration = 99999;
   fadeIn = 0.2;
   fadeOut = 1;
   idd = 2992;
   onLoad = "uiNamespace setVariable ['RscDisplayBaseAttack', _this select 0]; ExecVM 'VEMFr_client\sqf\RscDisplayBaseAttack\onLoad.sqf'";
   onUnLoad = "uiNamespace setVariable ['RscDisplayBaseAttack', displayNull]";

   class RscText
	{
		access = 0;
		colorBackground[] = {0.071, 0.024, 0.024, 0.75};
		colorShadow[] = {0, 0, 0, 0.5};
		colorText[] = {1, 0.333, 0.333, 1};
		deletable = 0;
		fade = 0;
		fixedWidth = 0;
		font = "PuristaMedium";
		h = 0.037;
		idc = -1;
		linespacing = 1;
		shadow = 1;
		sizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 1)";
		style = 2;
		text = "";
		tooltipColorBox[] = {1, 1, 1, 1};
		tooltipColorShade[] = {0, 0, 0, 0.65};
		tooltipColorText[] = {1, 1, 1, 1};
		type = 0;
		w = 0.3;
		x = 0;
		y = 0;
	};

   class controls
   {
      class text:RscText {
         x = 0.4 * safezoneW + safezoneX;
			y = 0.88 * safezoneH + safezoneY;
			w = 0.2 * safezoneW; //w = 0.35 * safezoneW;
			h = 0.0275 * safezoneH;

         font = "PuristaBold";
         idc = 1;
         shadow = 0;
         text = " // [ WARNING! Base is under attack ] \\";
      };
   };
};


VEMFr_client/gui/RscDisplayVEMFrClient.hpp
class RscDisplayVEMFrClient
{
	idd = 2991;
	fadeIn = 0.2;
	fadeOut = 1;
	duration = 99999;
	onLoad = "uiNamespace setVariable ['RscDisplayVEMFrClient', _this select 0]; ExecVM 'VEMFr_client\sqf\RscDisplayVEMFrClient\onLoad.sqf'";
	onUnLoad = "uiNamespace setVariable ['RscDisplayVEMFrClient', displayNull]";
	class RscBackground
	{
		access = 0;
		colorBackground[] = {0.48,0.5,0.35,1};
		colorShadow[] = {0,0,0,0.5};
		colorText[] = {0.1,0.1,0.1,1};
		deletable = 0;
		fade = 0;
		fixedWidth = 0;
		font = "PuristaLight";
		h = 1;
		idc = -1;
		linespacing = 1;
		shadow = 0;
		SizeEx = 1;
		style = 512;
		text = "";
		tooltipColorBox[] = {1,1,1,1};
		tooltipColorShade[] = {0,0,0,0.65};
		tooltipColorText[] = {1,1,1,1};
		type = 0;
		w = 1;
		x = 0;
		y = 0;
	};
	class RscPictureKeepAspect
	{
		access = 0;
		colorBackground[] = {0,0,0,1};
		colorText[] = {1,1,1,1};
		deletable = 0;
		fade = 0;
		fixedWidth = 0;
		font = "PuristaMedium";
		h = 0.15;
		idc = -1;
		lineSpacing = 0;
		shadow = 0;
		sizeEx = 0;
		style = "0x30 + 0x800";
		text = "";
		tooltipColorBox[] = {1,1,1,1};
		tooltipColorShade[] = {0,0,0,0.65};
		tooltipColorText[] = {1,1,1,1};
		type = 0;
		w = 0.2;
		x = 0;
		y = 0;
	};
	class RscText
	{
		access = 0;
		colorBackground[] = {0.071,0.078,0.094,1};
		colorShadow[] = {0,0,0,0.5};
		colorText[] = {0.22,0.745,0.882,1};
		deletable = 0;
		fade = 0;
		fixedWidth = 0;
		font = "PuristaMedium";
		h = 0.037;
		idc = -1;
		linespacing = 1;
		shadow = 1;
		sizeEx = "(			(			(			((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 1)";
		style = 0;
		text = "";
		tooltipColorBox[] = {1,1,1,1};
		tooltipColorShade[] = {0,0,0,0.65};
		tooltipColorText[] = {1,1,1,1};
		type = 0;
		w = 0.3;
		x = 0;
		y = 0;
	};
	class controls
	{
		class icon:RscPictureKeepAspect
		{
			x = 0.25 * safezoneW + safezoneX;
			y = 0.825 * safezoneH + safezoneY;
			w = 0.025 * safezoneW;
			h = 0.05 * safezoneH;

			fade = 1;
			idc = 10;
			text = "\A3\ui_f\data\map\markers\nato\o_art.paa";
		};
		class txtTitle: RscText
		{
			x = 0.2875 * safezoneW + safezoneX;
			y = 0.8235 * safezoneH + safezoneY;
			w = 0 * safezoneW;
			h = 0.02 * safezoneH;

			colorBackground[] = {0,0,0,0.75};
			colorText[] = {1,1,1,0.9};
			fade = 1;
			idc = 11;
			shadow = 0;
			sizeEx = "(			(			(			((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 0.85)";
			text = "";
		};
		class txtMsg: RscText
		{
			x = 0.2875 * safezoneW + safezoneX;
			y = 0.845 * safezoneH + safezoneY;
			w = 0 * safezoneW;
			h = 0.03 * safezoneH;

			colorBackground[] = {0,0,0,0.75};
			colorText[] = {1,1,1,0.9};
			fade = 1;
			idc = 12;
			shadow = 0;
			sizeEx = "(			(			(			((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 1.25)";
			text = "";
		};
	};
};
